packages/core/src/shared/telemetry/telemetryService.ts

```diff
@@ -195,9 +195,11 @@ export class DefaultTelemetryService {
     }
 
     /**
-     * Publish metrics to the Telemetry Service.
+     * Publish metrics to the Telemetry Service. Usually it will automatically flush recent events
+     * on a regular interval. This should not be used unless you are interrupting this interval,
+     * e.g. via a forced window reload.
      */
-    private async flushRecords(): Promise<void> {
+    public async flushRecords(): Promise<void> {
```

Comment:
we have a `didReload` concept https://github.com/aws/aws-toolkit-vscode-staging/blob/e02cdb442648177d4e08aab331a9379ad57f9345/packages/core/src/shared/extensionGlobals.ts#L151

which handles window reloads. And our telemetry client has a filesystem cache of telemetry events, which are found on vscode start/reload. 

Given the above, is it still necessary to expose `flushRecords` ?

Reply:
`didReload` applies to SAM only, as it checks for the existence of a key set by it. This also means it only applies to toolkit and not Q.

I see now that telemetry writes to a file on deactivation, previously it looked like it only wrote to an in-memory queue. I will need to verify that `deactivation` definitely runs on this forced window reload as expected.

Reply:
`deactivation` is not being called for a force reload. Very interesting. So, we still need this call.